core/src/test/java/org/elasticsearch/index/query/IndicesQueryBuilderTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class IndicesQueryBuilderTest extends BaseQueryTestCase<IndicesQueryBuilder> {
+
+    //NO COMMIT if we want to directly compare the inner queries
+    @Override
+    protected boolean supportsBoostAndQueryName() {
+        return false;
+    }
+
+    @Override
+    protected IndicesQueryBuilder doCreateTestQueryBuilder() {
+        IndicesQueryBuilder query = new IndicesQueryBuilder(
+                RandomQueryBuilder.createQuery(random()), generateRandomStringArray(5, 10, false, false));
+        
+        switch (randomInt(2)) {
+            case 0:
+                query.noMatchQuery(RandomQueryBuilder.createQuery(random()));
+                break;
+            case 1:
+                query.noMatchQuery(randomFrom(QueryBuilders.matchAllQuery(), QueryBuilders.matchNoneQuery()));
+                break;
+            default:
+                // do not set noMatchQuery
+        }
+        return query;
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(IndicesQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        if (context.matchesIndices(queryBuilder.indices())) {
+            assertEquals(query, queryBuilder.innerQuery().toQuery(context));
+        } else {
+            assertEquals(query, queryBuilder.noMatchQuery().toQuery(context));
```

Comment:
are we injecting indices to the test cluster service? or do we always test one branch only of these two?